### PR TITLE
build: Add SGX, VFIO and rate limit testing to MQ

### PR DESCRIPTION
**Pull request by rbradford**
Run these workflows as part of the merge queue to help improve testing
coverage.

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

### CODE REVIEW

**Comment by likebreath** [comment]
We will need to make these workers required, right?

**Review by likebreath — APPROVED** [review]

**Comment by likebreath** [comment]
@rbradford I added a commit to fix the directory permissions on the bare-metal workers. PTAL.

File: .github/workflows/integration-rate-limiter.yaml [hunk 1]
```diff
@@ -1,18 +1,28 @@
 name: Cloud Hypervisor Tests (Rate-Limiter)
 on:
+  merge_group:
+  pull_request:
   push:
     branches:
       - main
```
Comment (likebreath) [inline .github/workflows/integration-rate-limiter.yaml:7]: We don't need to run rate-limiter (and VFIO/SGX) on main branch any more, given we have it covered from merge queue. Right?

File: .github/workflows/integration-sgx.yaml [hunk 2]
```diff
@@ -1,21 +1,32 @@
 name: Cloud Hypervisor Tests (SGX)
 on:
+  merge_group:
+  pull_request:
   push:
     branches:
       - main
```
Comment (likebreath) [inline .github/workflows/integration-sgx.yaml:7]: Ditto

File: .github/workflows/integration-vfio.yaml [hunk 3]
```diff
@@ -1,21 +1,32 @@
 name: Cloud Hypervisor Tests (VFIO)
 on:
+  merge_group:
+  pull_request:
   push:
     branches:
       - main
```
Comment (likebreath) [inline .github/workflows/integration-vfio.yaml:7]: Ditto

**Comment by rbradford** [comment]
> @likebreath Oops, I missed your other comment about the permissions fix - can you create a new PR - I guess we need the same for ARM64?

No matter - I was able to find the commit and cherry-picked it.

**Pull request merged**